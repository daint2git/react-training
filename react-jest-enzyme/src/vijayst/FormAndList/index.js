import React from 'react'

import Form2 from '../Form2'
import List from '../List'

export default class FormAndList extends React.Component {
  state = { list: [] }

  handleAdd = name => {
    this.setState(prevState => ({ list: [...prevState.list, name] }))
  }

  render() {
    return (
      <div>
        <Form2 onAdd={this.handleAdd} />
        <List data={this.state.list} />
      </div>
    )
  }
}
