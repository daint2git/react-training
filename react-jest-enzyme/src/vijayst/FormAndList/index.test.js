import { mount, shallow, render } from 'enzyme'
import renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import FormAndList from './'

describe('FormAndList', () => {
  describe('Snapshot', () => {
    test('1. renderer', () => {
      const component = renderer.create(<FormAndList />)
      const json = component.toJSON()
      expect(json).toMatchSnapshot()
    })

    test('2. shallow renderer', () => {
      const renderer = new ShallowRenderer()
      renderer.render(<FormAndList />)
      const json = renderer.getRenderOutput()
      expect(json).toMatchSnapshot()
    })

    test('3. shallow', () => {
      const wrapper = shallow(<FormAndList />)
      expect(wrapper).toMatchSnapshot()
    })

    test('4. mount', () => {
      const wrapper = mount(<FormAndList />)
      expect(wrapper).toMatchSnapshot()
    })

    test('5. mount (executed render method)', () => {
      const wrapper = mount(<FormAndList />)
      expect(wrapper.render()).toMatchSnapshot()
    })

    test('6. render', () => {
      const wrapper = render(<FormAndList />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('structure and behavior', () => {
    let wrapper

    beforeEach(() => {
      wrapper = shallow(<FormAndList />)
    })

    test('FormAndList renders nested components', () => {
      expect(wrapper.find('Form2').length).toEqual(1)
      expect(wrapper.find('List').length).toEqual(1)
    })

    test('onAdd updates List', () => {
      const inputValue = 'Email 1'
      const form = wrapper.find('Form2').first()
      expect(wrapper).toMatchSnapshot()
      form.props().onAdd(inputValue)
      expect(wrapper).toMatchSnapshot()
      const list = wrapper.find('List').first()
      const data = list.props().data
      expect(data.length).toEqual(1)
      expect(data[0]).toEqual(inputValue)
    })
  })
})
