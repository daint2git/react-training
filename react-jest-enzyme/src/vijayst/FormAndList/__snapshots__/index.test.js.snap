// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAndList Snapshot 1. renderer 1`] = `
<div>
  <form
    className="sc-bdVaJa ksQzqc"
  >
    <input
      className="sc-bwzfXH gmViiu"
      name="email"
      onChange={[Function]}
      type="text"
    />
    <button
      className="sc-htpNat keqZrr"
      onClick={[Function]}
    >
      Add
    </button>
  </form>
  <table
    className="myClass"
  >
    <thead>
      <tr>
        <th>
          Email
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`FormAndList Snapshot 2. shallow renderer 1`] = `
<div>
  <Form2
    onAdd={[Function]}
  />
  <List
    data={Array []}
  />
</div>
`;

exports[`FormAndList Snapshot 3. shallow 1`] = `
<div>
  <Form2
    onAdd={[Function]}
  />
  <List
    data={Array []}
  />
</div>
`;

exports[`FormAndList Snapshot 4. mount 1`] = `
<div>
  <styled.form>
    <form
      className="sc-bdVaJa ksQzqc"
    >
      <styled.input
        name="email"
        onChange={[Function]}
        type="text"
      >
        <input
          className="sc-bwzfXH gmViiu"
          name="email"
          onChange={[Function]}
          type="text"
        />
      </styled.input>
      <styled.button
        onClick={[Function]}
      >
        <button
          className="sc-htpNat keqZrr"
          onClick={[Function]}
        >
          Add
        </button>
      </styled.button>
    </form>
  </styled.form>
  <table
    className="myClass"
  >
    <thead>
      <tr>
        <th>
          Email
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`FormAndList Snapshot 5. mount (executed render method) 1`] = `
<div>
  <form
    class="sc-bdVaJa ksQzqc"
  >
    <input
      class="sc-bwzfXH gmViiu"
      name="email"
      type="text"
      value=""
    />
    <button
      class="sc-htpNat keqZrr"
    >
      Add
    </button>
  </form>
  <table
    class="myClass"
  >
    <thead>
      <tr>
        <th>
          Email
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`FormAndList Snapshot 6. render 1`] = `
<div>
  <form
    class="sc-bdVaJa ksQzqc"
  >
    <input
      class="sc-bwzfXH gmViiu"
      name="email"
      type="text"
    />
    <button
      class="sc-htpNat keqZrr"
    >
      Add
    </button>
  </form>
  <table
    class="myClass"
  >
    <thead>
      <tr>
        <th>
          Email
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`FormAndList structure and behavior onAdd updates List 1`] = `
<div>
  <Form2
    onAdd={[Function]}
  />
  <List
    data={Array []}
  />
</div>
`;

exports[`FormAndList structure and behavior onAdd updates List 2`] = `
<div>
  <Form2
    onAdd={[Function]}
  />
  <List
    data={
      Array [
        "Email 1",
      ]
    }
  />
</div>
`;
